### ISSUE TITLE
Memory leak with nest.js custom server

### ISSUE THREAD

**Issue by elrion018** [vercel/next.js#37298]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Enterprise
    Binaries:
      Node: 14.17.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.1.7-canary.23
      react: 17.0.2
      react-dom: 17.0.2

### What browser are you using? (if relevant)

Chrome 102.0.5005.62

### How are you deploying your application? (if relevant)

AWS deploy with Jenkins

### Describe the Bug

I am using next.js with a custom server made of nest.js(@nestjs/core": "8.0.0). A request is received from the controller of nest.js, but I have configured next.js to proceed with rendering of the request.

I think this issue is similar with https://github.com/vercel/next.js/issues/10461

`app.container.ts`
```
export class AppContainer extends BaseAppContainer {
  public listen(app: NestExpressApplication): Promise<void> {
    /** @see {https://docs.nestjs.com/techniques/validation} **/
    app.useGlobalPipes(new ValidationPipe({ transform: true }));

    const configService = app.get(ConfigService);
    const port = Number(configService.get('port')) || 8080;
    return app.listen(port, () => {

    });
  }
}

```

Detailed logic has been implemented in `view.service.ts`

`view.service.ts`
```
import { Injectable, OnModuleInit } from '@nestjs/common';
import { Response, Request } from 'express';
import next from 'next';

@Injectable()
export class ViewService implements OnModuleInit {
  private server: any;

  async onModuleInit(): Promise<void> {
    try {
      this.server = next({ dev: false, dir: './' });

      await this.server.prepare();
    } catch (error) {
      console.error(error);
    }
  }

  handler(req: Request, res: Response) {
    return this.server.getRequestHandler()(req, res);
  }
}
```

The user's request arrives at view.controller.ts.

```
import { Controller, Get, Req, Res } from '@nestjs/common'

import { ViewService } from './views.service';
import { Request, Response } from 'express';

@Controller('/')
export class ViewsController {
  constructor(
    private homeDataService: HomeDataService,
    private viewService: ViewService
  ) {
    this.viewService = viewService;
  }

  @Get('_next*')
  async static(@Req() req: Request, @Res() res: Response) {
    return await this.viewService.handler(req, res);
  }

  @Get('static/favicon.ico')
  async favicon(@Req() req: Request, @Res() res: Response) {
    return await this.viewService.handler(req, res);
  }

  @Get('/')
  async feed(@Req() req: Request, @Res() res: Response) {
    return await this.viewService.handler(req, res);
  }
}
```

I run a service that has about 400,000 visits per day. 

When there is a lot of traffic, memory occupancy increases rapidly.

![memory_leak](https://user-images.githubusercontent.com/41279460/170974148-e264490d-c6ec-4567-85ab-52cc9f91f980.PNG)

This was tested locally by continuously pressing the refresh button by hand.
![heapdump1](https://user-images.githubusercontent.com/41279460/170974659-f1611584-f6e0-4f6c-bf22-d38a911363f9.PNG)
![heapdump2](https://user-images.githubusercontent.com/41279460/170975411-695e95c8-b98c-4673-9a76-f8b55848f552.PNG)

`Generator` , `Url` ,`NodeNextRequest` , `NodeNextResponse` : As a result of watching over time, these are never garbage collected.

Heapdump files are provided : 

[heapdumps.zip](https://github.com/vercel/next.js/files/8797821/heapdumps.zip)

### Expected Behavior

No memory leaks and smooth memory occupancy

### To Reproduce

1. Write `app.container.ts`.
2. Write `view.service.ts`.
3. Write `view.controller.ts`.
4. `cross-env node .next/production-server/src/main.js`
5. You have to put a lot of load on the server simultaneously.

**Comment by timneutkens** [comment]
Can you use run the app without adding this overhead of additional libraries. It's unclear if the additional layer you've added could be causing this. Potentially it's related to how this could be initialized many times:
```
try {
      this.server = next({ dev: false, dir: './' });

      await this.server.prepare();
    } catch (error) {
      console.error(error);
    }
```

It might be helpful to instantiate it only once.

**Comment by elrion018** [comment]
@timneutkens 

The code snippet part you suggested will be initialized once as a result of adding logging. When coding at first, I had this part in mind and coded.

Additional debugging results show that when the controller serving static data is annotated, garbage memories are not accumulated.

```
@Controller('/')
export class ViewsController {
  constructor(
    private homeDataService: HomeDataService,
    private viewService: ViewService
  ) {
    this.homeDataService = homeDataService;
    this.viewService = viewService;
  }

  // @Get('_next*')
  // async static(@Req() req: Request, @Res() res: Response) {
  //   await this.viewService.handler(req, res);
  // }

  // @Get('static/favicon.ico')
  // async favicon(@Req() req: Request, @Res() res: Response) {
  //   await this.viewService.handler(req, res);
  // }

  @Get('/')
  async feed(@Req() req: Request, @Res() res: Response) {
    await this.viewService.handler(req, res);
  }
}


```

**Comment by JanKaifer** [comment]
Hey, memory leaks are really nasty to catch.
Could you help us by trying o create a simple reproduction? Ideally a clonable repo.

**Comment by balazsorban44** [comment]
This issue has been automatically closed because it received no activity for a month and had no reproduction to investigate. If you think it was closed by accident, please leave a comment. If you are running into a similar issue, please open a new issue with a reproduction. Thank you.
